src/Utils/Responsive/Media.tsx

```diff
@@ -0,0 +1,268 @@
+import PropTypes from "prop-types"
+import React from "react"
+import styled, { css, InterpolationValue } from "styled-components"
+
+const MutuallyExclusiveProps = [
+  "query",
+  "at",
+  "lessThan",
+  "greaterThan",
+  "greaterThanOrEqual",
+  "between",
+  "interaction",
+]
+
+// TODO: All of these props should be mutually exclusive. Using a union should
+//       probably be made possible by https://github.com/Microsoft/TypeScript/pull/27408.
+export interface Props<B, I> {
```

Comment:
Instead of this, could we use conditional types? Here's an example that (I think) is doing what we want. http://realfiction.net/2018/06/10/ts-conditional-types/

Reply:
Eh, nope. Scratch that. Isn't what we want. >_>

Reply:
Wait, why not? I haven’t read extensively, but to me it does appear to do the thing we want.